stdlib/pervasives.ml

```diff
@@ -294,6 +294,17 @@ external open_desc : string -> open_flag list -> int -> int = "caml_sys_open"
 external set_out_channel_name: out_channel -> string -> unit =
   "caml_ml_set_channel_name"
 
+(* [finally_ f g x] calls [f x], but takes care of always calling [g x]
+   whether [f] raises or returns *)
+let finally_ f g x =
+  try
+    let res = f x in
+    g x;
+    res
+  with e ->
+    g x;
+    raise e
+
 let open_out_gen mode perm name =
```

Comment:
This does not have the right semantics if `g` raises. Better to use
```ocaml
match f x with
| res -> g x;  res
| exception e -> g x; raise e
```

Reply:
The same `finally_` is defined in `otherlibs/threads/pervasives.ml` as well. I think it would be better to define it just once in pervasives and expose it, then you can call it in `threads` as well.